### PR TITLE
added code for coins problem of codechef in C++

### PR DESCRIPTION
**Pull request by abhijeet010304**
**Fixes issue:**

Created code for COINS problem of codechef in C++

**Changes:**

Added code for COINS problem of codechef in C++

### CODE REVIEW

File: code/online_challenges/src/codechef/COINS/coins.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include<bits/stdc++.h>
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/coins.cpp]: #1915. Remove using statement. Reformat this file using clang-format with 4 spaces.

File: code/online_challenges/src/codechef/COINS/coins.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        mpp[n]=check(n/2)+check(n/3)+check(n/4);
+        mpp[n]=max(n,mpp[n]);
+        return mpp[n];
+      }
+}
+int main()
+{
+
+ 	lli  n;
+ 	while(cin>>n)
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/coins.cpp]: Can use `for (lli n; std::cin >> n;)`.

File: code/online_challenges/src/codechef/COINS/coins.cpp (hunk 1 above)
Comment (abhijeet010304) [reply code/online_challenges/src/codechef/COINS/coins.cpp]: OK..

File: code/online_challenges/src/codechef/COINS/coins.cpp (hunk 2 above)
Comment (abhijeet010304) [reply code/online_challenges/src/codechef/COINS/coins.cpp]: OK sir I will change it too

Comment (abhijeet010304) [reply code/online_challenges/src/codechef/COINS/coins.cpp]: Done sir please review changes.

File: code/online_challenges/src/codechef/COINS/coins.cpp (hunk 1 above)
Comment (abhijeet010304) [reply code/online_challenges/src/codechef/COINS/coins.cpp]: Sir I have done changes please review them.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
+#include<iostream>
+#include<malloc.h>
+using namespace std;
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: Don;t use this.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
+#include<iostream>
+#include<malloc.h>
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: You shouldn't need this.

File: code/online_challenges/src/codechef/COINS/coins.cpp (hunk 2 above)
Comment (abhijeet010304) [reply code/online_challenges/src/codechef/COINS/coins.cpp]: and sir I have also added code for implementation of stack using linked list , its not shown as another PR ,please guide me what to do now

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+using namespace std;
+struct node
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: Capital `Node`.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+{
+    int data;
+    struct node *next;
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: struct prefix not needed in C++.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+};
+void Traverse_stack(struct node* top)
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: fix names. Should all be lower camel case.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+{
+    struct node* temp=top;
+    cout<<"Current Stack->\n";
+    while(temp!=NULL)
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: nullptr.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    {
+    cout<<temp->data<<"\n";
+    temp=temp->next;
+    }
+}
+struct node * push(struct node * top,int data)
+{
+    struct node * temp;
+    temp=(struct node*)malloc(sizeof(struct node));
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: Use new and delete in C++.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    temp->data=data;
+    temp->next=top;
+    top=temp;
+    cout<<top->data<<" is pushed into the stack\n";
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: Don't output in operations, only when explicitly printing.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 4 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: oh means i should use (new ) keyword to allocate memory

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    Traverse_stack(top);
+    return top;
+}
+struct node* pop(struct node* top)
+{
+    struct node * temp;
+    temp=top;
+    if(top==NULL)
+    {
+            cout<<"Stack is empty\n";
```
Comment (arnavb) [inline code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: Throw an exception instead.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 5 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: okk..

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 6 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: okay ..

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 7 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: okay..

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 8 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: both are same I think while(temp!=NULL) and while(!temp)

Comment (arnavb) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: They are, but don't use NULL in C++ if you want to be explicit, use nullptr instead.

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 9 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: ha i got it.. sir I think I should commit the same code in C language also and also make changes and submit the modified code in c++

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 10 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: OK..

File: code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp (hunk 11 above)
Comment (abhijeet010304) [reply code/data_structures/src/stack/stackUsingLinkedList/stackUsingLinkedList.cpp]: OK..

**Comment by arnavb** [comment]
@abhijeet010304 Regarding your combined PRs it's because you didn't use separate branches when creating separate changes. You can learn about branches in git. For now, I'll merge all your changes in this PR and you can close your other pull requests.

**Comment by abhijeet010304** [comment]
sir I think i should remove the code of stack implementation from the PR since that would take some time again in another PR I will implement that with all the changes

File: code/online_challenges/src/codechef/COINS/coins.cpp [hunk 12]
```diff
@@ -0,0 +1,34 @@
+#include<iostream>
+#include <unordered_map>
+using namespace std ;
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/coins.cpp]: Don't use this.

**Comment by abhijeet010304** [comment]
Done sir now you can merge the previous COINS problem and I will start working on stack problem now

File: code/online_challenges/src/codechef/COINS/coins.cpp [hunk 13]
```diff
@@ -0,0 +1,34 @@
+#include<iostream>
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/coins.cpp]: REformat this file using clang format with 4 spaces.

**Comment by arnavb** [comment]
Once you remove the using namespace std and replace cout with std::cout I will merge.

**Comment by abhijeet010304** [comment]
Okay now I got that !! hihi...

File: code/online_challenges/src/codechef/COINS/coins.cpp [hunk 14]
```diff
@@ -0,0 +1,30 @@
+#include <iostream>
+#include <unordered_map>
+#define lli long long int
+std::unordered_map<lli, lli> mpp;
+lli check(lli n)
+{
+
```
Comment (arnavb) [inline code/online_challenges/src/codechef/COINS/coins.cpp]:
```suggestion
```

**Comment by abhijeet010304** [comment]
thanks a lot now I will start working on that stack problem!! hihi

**Pull request merged**